components/form/TaskForm.tsx

```diff
@@ -65,63 +169,25 @@ export function CreateTaskForm({
-                    <FormLabel>Task Name</FormLabel>
-                    <Input
-                        placeholder="Task Name"
-                        onChange={(event) =>
-                            setTaskFields({
-                                ...taskFields,
-                                name: event.target.value,
-                            })
-                        }
-                    />
-                    <FormErrorMessage>
-                        Task name cannot be empty.
-                    </FormErrorMessage>
-                </FormControl>
-                <FormControl>
-                    <FormLabel>Description</FormLabel>
-                    <Input
-                        placeholder="Description"
-                        onChange={(event) =>
-                            setTaskFields({
-                                ...taskFields,
-                                description: event.target.value,
-                            })
-                        }
-                    />
-                </FormControl>
-                <FormControl py="0.5rem">
-                    <Flex flexDirection={"row"} alignItems={"baseline"}>
-                        <FormLabel>Billable</FormLabel>
-                        <Checkbox
-                            isChecked={taskFields.billable}
-                            onChange={(event) =>
-                                setTaskFields({
-                                    ...taskFields,
-                                    billable: event.target.checked,
-                                })
-                            }
-                        />
-                    </Flex>
-                </FormControl>
-            </div>
-            <div style={{ textAlign: "right", padding: "20px" }}>
-                <Button colorScheme="blue" mr={3} onClick={onSubmit}>
-                    Submit
-                </Button>
-                <Button colorScheme="gray" onClick={onCancel}>
-                    Cancel
-                </Button>
-            </div>
-            {errorMessage && (
-                <>
-                    <ErrorAlert />
-                    <Box>{errorMessage}</Box>
-                </>
-            )}
+            <FormContainer>
+                <NameInput
+                    taskFields={taskFields}
+                    setTaskFields={setTaskFields}
+                />
+                <DescriptionInput
+                    taskFields={taskFields}
+                    setTaskFields={setTaskFields}
+                />
+                <BillableInput
+                    taskFields={taskFields}
+                    setTaskFields={setTaskFields}
+                />
+            </FormContainer>
+            <FormContainer>
+                <FromSubmitButton onClick={onSubmit} />
+                <FromCancelButton onClick={onCancel} />
+            </FormContainer>
+            {errorMessage && <FormAlerts errorMessage={errorMessage} />}
```

Comment:
The form looks pretty nice!

However, the components seem to be operating on a suboptimal level of abstraction. A component
```typescript
<DescriptionInput
      taskFields={taskFields}
      setTaskFields={setTaskFields}
/>
```
is not very recyclable, since it's only relevant if the form contains a field `description`.

I think the correct level of abstraction is something along the lines of
```typescript
<FormInputField
    label={”Description"}
    placeholder={"Description"}
    onChange={(e) => setTaskFields({...taskFields, description: e.target.value})}

... etc etc

 />
```
I.e. I think we should make a general "FormInputField" component that is defined once and can be used in all forms.

I'd make a separate component for the chechbox.

So in this form we should have two input fields and one checkbox field.